app/controllers/users/passwords_controller.rb

```diff
@@ -55,10 +60,31 @@ def handle_valid_password
         user_session[:personal_key] = @update_user_password_form.personal_key
         redirect_to manage_personal_key_url
       else
-        redirect_to account_url
+        redirect_to account_or_after_sign_in_path
```

Comment:
Oh, I see the reason we're assigning an instance variable is partly to make sure it's available in `account_or_after_sign_in_path` below after the session value is deleted.

Maybe we could revise that to avoid relying on the instance variable side-effect before deleting the session value?

```rb
def handle_valid_password
  send_password_reset_risc_event
  # ...
  if @update_user_password_form.personal_key.present?
    # ...
  elsif required_password_change?
    session.delete(:redirect_to_change_password)
    redirect_to after_sign_in_path_for(current_user)
  else
    redirect_to account_path
  end
end
```